File: src/main/java/rk/cinema/repository/SeatRepository.java
package rk.cinema.repository;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.CrudRepository;
import rk.cinema.model.Seat;

import java.util.List;

@JdbcRepository(dialect = Dialect.POSTGRES)
public interface SeatRepository extends CrudRepository<Seat, Long> {
    @NonNull
    List<Seat> findByHallId(@NonNull Long hallId);
}

File: src/main/kotlin/rk/cinema/repository/SeatRepository.kt
package rk.cinema.repository

import io.micronaut.core.annotation.NonNull
import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect
import io.micronaut.data.repository.CrudRepository
import rk.cinema.model.Seat

@JdbcRepository(dialect = Dialect.POSTGRES)
interface SeatRepository : CrudRepository<Seat, Long> {
    fun findByHallId(hallId: @NonNull Long): @NonNull MutableList<Seat>
}
